core/clustersmngr/factory.go

```diff
@@ -258,13 +264,74 @@ func (cf *clustersManager) UpdateNamespaces(ctx context.Context) error {
 				continue
 			}
 
+			// set the namespaces for the cluster
 			cf.clustersNamespaces.Set(clusterName, list.Items)
 		}
+
+		// launch the namespace watcher for this cluster
+		if configs[clusterName] != nil {
+			go func(ctx context.Context, restConfig *rest.Config, clusterName string, errChan chan error) {
+				cf.updateNamespaceWithWatch(ctx, restConfig, clusterName, errChan)
+			}(ctx, configs[clusterName], clusterName, errChan)
```

Comment:
Similar pattern coming together here https://github.com/weaveworks/weave-gitops/pull/2710

Reply:
yes, when we deal we mutliple cluster we would have to watch the clusters updates and act accordingly. IMO we should not have to deal with it in this code, because we have only a single cluster in the OSS version.

Reply:
Yeah, good thoughts. EE is using this module pretty much as is right now. EE provides a different [`Fetch()`](https://github.com/weaveworks/weave-gitops/blob/3977758e0841e1c88579a209021079a203723ff0/core/clustersmngr/fetcher/single.go#L20-L21) method to the core one here, but thats it. There's not really any other Multi-cluster (MC) querying code in EE.

All of the Multiple-cluster (MC) querying code is here in core. Refactoring this and moving the MC responsibility out of core would require an interface redesign and quite some thought.

To keep this keep this PR scoped to `watch`ing it might be easiest to handle cluster add/removal here instead of the larger refactor/redesign as far as I understand it.